supabase/pg_dump.sql

```diff
@@ -112,6 +1546,44 @@ end;$$;
 
 ALTER FUNCTION "public"."create_chat"() OWNER TO "postgres";
 
+--
+-- Name: create_drives_from_recurring(); Type: FUNCTION; Schema: public; Owner: postgres
+--
+
+CREATE FUNCTION "public"."create_drives_from_recurring"() RETURNS "void"
+    LANGUAGE "plpgsql"
+    AS $$
+DECLARE
+	recurring_drive RECORD;
+	drive_start_time TIMESTAMP;
+	drive_end_time TIMESTAMP;
+BEGIN
+	FOR recurring_drive IN
+		SELECT
+			*
+		FROM
+			recurring_drives
+		WHERE
+			_rrule.rruleset (recurrence_rule) @> (now()::timestamp + interval '30 days') AND
+			stopped_at IS NULL
+	LOOP
+		drive_start_time = CURRENT_DATE + interval '30 days' + recurring_drive.start_time;
```

Comment:
Can you refactor this into a function?

Reply:
I certainly can, the question is whether it is a good idea. The function would look like this

```
CREATE OR REPLACE FUNCTION public.recurring_drive_generate_datetimes(base_date date, start_time time without time zone, end_time time without time zone)
 RETURNS record
 LANGUAGE plpgsql
 IMMUTABLE
AS $function$
DECLARE
	ret RECORD;
	generated_start_time TIMESTAMP;
	generated_end_time TIMESTAMP;
BEGIN
   	generated_start_time = base_date + start_time;
	generated_end_time = base_date + end_time;
	IF start_time > end_time THEN
		generated_end_time = generated_end_time + interval '1 day';
	END IF;
	
	SELECT generated_start_time, generated_end_time INTO ret;
	
	RETURN ret;	
END; $function$
```

and then it would be called like this
```
SELECT generated_start_time, generated_end_time 
INTO drive_start_time, drive_end_time 
FROM recurring_drive_generate_datetimes(base_date, recurring_drive.start_time, recurring_drive.end_time) 
AS ret (generated_start_time TIMESTAMP, generated_end_time TIMESTAMP);
```

That is compared to this code that was used before
```
drive_start_time = next_date + recurring_drive.start_time;
drive_end_time = next_date + recurring_drive.end_time;
IF recurring_drive.start_time > recurring_drive.end_time THEN
	drive_end_time = drive_end_time + interval '1 day';
END IF;
```

Reply:
Well now that I have already written it, I will (tentatively) add it

Reply:
What I had in mind was to include the INSERT in the function as well